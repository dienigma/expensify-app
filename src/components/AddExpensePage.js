import React from "react";

const AddExpensePage = () => (
  <div>This is from the Add Expense Page component</div>
);
export default AddExpensePage;
