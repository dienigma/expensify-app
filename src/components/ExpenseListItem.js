import React from "react";
import { connect } from "react-redux";
import { Link } from "react-router-dom";
import { removeExpense } from "../actions/expenses";

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  return (
    <div>
      <h3>
        <Link to={`/edit/${id}`}>{description}</Link>
      </h3>
      <p>
        {amount}-{createdAt}
      </p>
    </div>
  );
};
export default ExpenseListItem;
