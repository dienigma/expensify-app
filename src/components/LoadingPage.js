import React from "react";

const LoadingPage = () => {
  return (
    <div className="loader">
      <img className="loader__image" src="/images/custom-loader.gif" />
    </div>
  );
};

export default LoadingPage;
