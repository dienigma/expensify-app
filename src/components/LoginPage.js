import React from "react";
import { connect } from "react-redux";
import { startLogin } from "../actions/auth";

export const LoginPage = ({ startLogin }) => {
  return (
    <div className="box-layout">
      <div className="box-layout__box">
        <h1 className="box-layout__title">Expensify</h1>
        <p>A way to get yourself together :)</p>
        <button onClick={startLogin} className="button">
          Login with google
        </button>
      </div>
    </div>
  );
};

const mapDisptachToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(
  undefined,
  mapDisptachToProps
)(LoginPage);
