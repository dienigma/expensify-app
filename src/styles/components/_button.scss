.button {
  color: white;
  background: $dark-blue;
  border: none;
  border-radius: 10px;
  font-size: $font-size-large;
  font-weight: 300;
  padding: $s-size;
  line-height: 1;
  display: inline-block;
  text-decoration: none;
  outline: none;
  &--link {
    background: none;

    color: #333;
  }
  &--secondary {
    background: #888;
  }
}
