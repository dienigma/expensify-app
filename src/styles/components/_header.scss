.header {
  background: $soft-yellow;
  &__title {
    color: #333;
    text-decoration: none;
    h1 {
      margin: 0;
    }
  }
  &__content {
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: $s-size 0;
  }
}
