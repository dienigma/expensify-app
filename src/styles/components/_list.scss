.list-header {
  background: $light-grey;
  border: 1px solid darken($light-grey, 7);
  color: $regular-grey;
  display: flex;
  justify-content: space-between;
  padding: $s-size $m-size;
}

.list-item {
  border: 1px solid darken($light-grey, 7);
  border-top: none;
  color: $regular-grey;
  display: flex;
  flex-direction: column;
  padding: $s-size;
  text-decoration: none;
  transition: background 0.3s ease;
  &:hover {
    background: $light-grey;
  }
  @media (min-width: $desktop-breakpoint) {
    align-items: center;
    flex-direction: row;
    justify-content: space-between;
    padding: $m-size;
  }
  &__title {
    margin: 0;
    word-break: break-all;
  }
  &__subtitle {
    color: $regular-grey;
    font-size: $font-size-small;
  }
  &__data {
    margin: 0 0 0 $s-size;
    @media (min-width: $desktop-breakpoint) {
      margin: 0;
      padding-left: $s-size;
    }
  }
  &--message {
    align-items: center;
    justify-content: center;
    color: $regular-grey;
    padding: $m-size;
    &:hover {
      background: none;
    }
  }
}

.list-body {
  margin-bottom: $m-size;
  @media (min-width: $desktop-breakpoint) {
    margin-bottom: $l-size;
  }
}
