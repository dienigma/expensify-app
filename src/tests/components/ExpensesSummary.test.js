import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("should render the component", () => {
  const wrapper = shallow(<ExpensesSummary />);
  expect(wrapper).toMatchSnapshot();
});

test("should render correctly with one expense", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={300} />
  );
  expect(wrapper).toMatchSnapshot();
});

test("should render correctly with multiple expenses", () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={24} expensesTotal={30890} />
  );
  expect(wrapper).toMatchSnapshot();
});
