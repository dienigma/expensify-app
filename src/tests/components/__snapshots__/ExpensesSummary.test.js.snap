// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    24
     
    expenses
     totalling 
    $308.90
  </h1>
</div>
`;

exports[`should render correctly with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $3.00
  </h1>
</div>
`;

exports[`should render the component 1`] = `
<div>
  <h1>
    Viewing 
     
    expenses
     totalling 
    $0.00
  </h1>
</div>
`;
